x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -137,50 +133,72 @@ public static String unquoteString(Source source) {
-            String replacement;
-
-            switch (group) {
-                case "\\t":
-                    replacement = "\t";
-                    break;
-                case "\\b":
-                    replacement = "\b";
-                    break;
-                case "\\f":
-                    replacement = "\f";
-                    break;
-                case "\\n":
-                    replacement = "\n";
-                    break;
-                case "\\r":
-                    replacement = "\r";
-                    break;
-                case "\\\"":
-                    replacement = "\"";
-                    break;
-                case "\\'":
-                    replacement = "'";
-                    break;
-                case "\\\\":
-                    // will be interpreted as regex, so we have to escape it
-                    replacement = "\\\\";
-                    break;
-                default:
-                    // unknown escape sequence, pass through as-is
-                    replacement = group;
+        StringBuilder sb = new StringBuilder();
+
+        for (int i = 0; i < text.length();) {
+            if (text.charAt(i) == '\\') {
+                // ANTLR4 Grammar guarantees there is always a character after the `\`
+                switch (text.charAt(++i)) {
+                    case 't':
+                        sb.append('\t');
+                        break;
+                    case 'b':
+                        sb.append('\b');
+                        break;
+                    case 'f':
+                        sb.append('\f');
+                        break;
+                    case 'n':
+                        sb.append('\n');
+                        break;
+                    case 'r':
+                        sb.append('\r');
+                        break;
+                    case '"':
+                        sb.append('\"');
+                        break;
+                    case '\'':
+                        sb.append('\'');
+                        break;
+                    case 'u':
+                        i = handleUnicodeChars(source, sb, text, ++i);
+                        break;
+                    case '\\':
+                        sb.append('\\');
+                        // will be interpreted as regex, so we have to escape it
+                        break;
+                    default:
+                        // unknown escape sequence, pass through as-is, e.g: `...\w...`
+                        sb.append('\\').append(text.charAt(i));
+                }
+                i++;
+            } else {
+                sb.append(text.charAt(i++));
             }
+        }
+        return sb.toString();
+    }
 
-            regexMatcher.appendReplacement(resultString, replacement);
-
+    private static int handleUnicodeChars(Source source, StringBuilder sb, String text, int i) {
+        String unicodeSequence;
+        int startIdx = i + 1;
```

Comment:
looks like here it assumes the first character is `{` but it's not enforced by the grammar

Reply:
It is enforced, check: https://github.com/elastic/elasticsearch/pull/70514/files#diff-9b0fe07d974e767795fe949863d7ae0b9d13787854b1435cba514335b75e62e5R215

Reply:
yeah you're right. sorry about that

Reply:
No worries!